client/scripts/components/Message/Compose.js

```diff
@@ -16,18 +16,21 @@ class Compose extends Component {
     super(props)
     this.state = {
       message: '',
-      showEmoji: false
+      showEmoji: false,
+      lastTypingPrompt: 0
     }
 
     this.onChange = (e) => {
       const text = e.target.value
       const {onUpdateText, showTypingPrompt, myNametag, roomId} = this.props
+      const {lastTypingPrompt} = this.state
       if (onUpdateText) {
         onUpdateText(text)
       }
 
-      if (showTypingPrompt) {
-        //TODO: Avoid posting on every keystroke
+      // Post a prompt that the user is typing at most once every 2 seconds
+      if (showTypingPrompt && lastTypingPrompt < Date.now() - 2000) {
+        this.setState({lastTypingPrompt: Date.now()})
```

Comment:
`setState` is not always synchronous, so I'm not sure if this will always update `lastTypingPrompt` soon enough (probably will 99.99% of the time, though). Another option would be to use `_.throttle` on a `this.showTypingPrompt` instance method.